packages/build-utils/src/fs/download.ts

```diff
@@ -92,12 +93,28 @@ export default async function download(
         await removeFile(basePath, name);
         return;
       }
-
       // If a file didn't change, do not re-download it.
       if (Array.isArray(filesChanged) && !filesChanged.includes(name)) {
         return;
       }
 
+      // Some builders resolve symlinks and return both
+      // a file, node_modules/<symlink>/package.json, and
+      // node_modules/<symlink>, a symlink.
+      // Removing the file matches how the yazl lambda zip
+      // behaves so we can use download() with `vercel build`.
+      const parts = name.split('/');
+      for (let i = 0; i < parts.length; i++) {
```

Comment:
```suggestion
      for (let i = 1; i < parts.length; i++) {
```

Minor optimization: I think we want to start at index 1 otherwise we'll always check an empty string entry.